### PR TITLE
[endpoint-test] Add test CLI mode

### PR DESCRIPTION
**Pull request by jaylees14**
I'm going to do this one in reverse from how we would normally - starting with the CLI and then adding more features from there.

Not sure I like `-g` for `--generated`, thoughts?

### CODE REVIEW

File: src/main/scala/temple/test/EndpointTester.scala [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package temple.test
+
+import temple.ast.Templefile
+
+object EndpointTester {
+
+  def test(templefile: Templefile, generatedPath: String): Unit =
+    println("Testing templefile...")
```
Comment (c-harding) [inline src/main/scala/temple/test/EndpointTester.scala:8]: Best test

File: src/main/scala/temple/Main.scala [hunk 2]
```diff
@@ -16,6 +16,7 @@ object Main extends App {
     val config = new TempleConfig(args)
     config.subcommand match {
       case Some(config.Generate) => Application.generate(config)
+      case Some(config.Test)     => Application.test(config)
```
Comment (c-harding) [inline src/main/scala/temple/Main.scala:18]: I predict a conflict (plz merge dev in)

**Review by Lewiky — APPROVED** [review]
LGTM - could use `-d` and `--dir` for the generated directory?

**Review by c-harding — APPROVED** [review]

File: src/main/scala/temple/test/EndpointTester.scala (hunk 1 above)
Comment (willjvsmith) [reply src/main/scala/temple/test/EndpointTester.scala:8]: The best test

**Review by willjvsmith — APPROVED** [review]

**Pull request merged**